Add livenessProbe rewrite example (#13224)
istio/istio.io: content/en/docs/ops/configuration/mesh/app-health-check/index.md

Istio solves both these problems by rewriting the application `PodSpec` readiness/liveness probe,
so that the probe request is sent to the [sidecar agent](/docs/reference/commands/pilot-agent/).

> 

Reviewer: If you add a section name here, the snippets will use the section name instead of just snip_x. Might make the test easier to read and maybe help the doc reader.

Author: agree

Author: agree

> The rewrite is visible at the application pod level. For example, using the [liveness-http-same-port sample]({{< github_file >}}/samples/health-check/liveness-http-same-port.yaml) sample, we see it working. First, create a namespace:

Reviewer: ```suggestion
The rewrite is visible at the application pod level. For example, using the [liveness-http-same-port sample]({{< github_file >}}/samples/health-check/liveness-http-same-port.yaml), we see it working. First, create and label a namespace:
```

Reviewer: Another option would be to number the tasks.

Istio solves both these problems by rewriting the application `PodSpec` readiness/liveness probe,
so that the probe request is sent to the [sidecar agent](/docs/reference/commands/pilot-agent/).

## Liveness probe rewrite example

> The rewrite is visible at the application pod level. For example, using [liveness-http-same-port sample]({{< github_file >}}/samples/health-check/liveness-http-same-port.yaml), we see it working. First, create and label a namespace:

Reviewer 2: ```suggestion
To demonstrate how the readiness/liveness probe is rewritten at the application PodSpec level, let us use the [liveness-http-same-port sample]({{< github_file >}}/samples/health-check/liveness-http-same-port.yaml).

First create and label a namespace for the example:
```

{{< text bash >}}

> $ kubectl create namespace istio-io-health-rewrite; kubectl label namespace istio-io-health-rewrite istio-injection=enabled

Reviewer 3: is there any sepical reason to create another namespace?

Reviewer 3: never mind

> $ kubectl create namespace istio-io-health-rewrite; kubectl label namespace istio-io-health-rewrite istio-injection=enabled

Reviewer 2: ```suggestion
$ kubectl create namespace istio-io-health-rewrite
$ kubectl label namespace istio-io-health-rewrite istio-injection=enabled
```

{{< /text >}}

> And deploy the sample app:

Reviewer 2: ```suggestion
And deploy the sample application:
```
